File: ios/Kurir/dependency/login/LoginModule.swift
//
//  LoginModule.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class LoginModule: Module {
    override func inject() {
        // ViewModel
        register(LoginViewModel.self) { r in
            LoginViewModel(r.resolve(AuthRepository.self)!)
        }
        
        // Delegate
        register(LoginDelegate.self) { (r, view: LoginView) -> LoginDelegate in
            LoginDelegate(view: view, viewModel: r.resolve(LoginViewModel.self)!)
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/login/impl/LoginModule.kt
package au.com.crazybean.mobilex.kurir.modules.auth.login.impl

import au.com.crazybean.mobilex.kurir.modules.auth.login.LoginDelegate
import au.com.crazybean.mobilex.kurir.modules.auth.login.LoginViewModel
import au.com.crazybean.mobilex.kurir.modules.auth.login.LoginView
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val loginModule = module {
    // Delegate
    factory { (view: LoginView?) ->
        LoginDelegate(view, get())
    }

    // ViewModel
    viewModel {
        LoginViewModel(get())
    }
}